doc/api/tls.md

```diff
@@ -1230,11 +1277,31 @@ changes:
     where `ctx` is a `SecureContext` instance. (`tls.createSecureContext(...)`
     can be used to get a proper `SecureContext`.) If `SNICallback` wasn't
     provided the default callback with high-level API will be used (see below).
+  * `pskCallback` {Function}
+     * identity: {string} identity parameter sent from the client.
+     * Returns: {Buffer|TypedArray|DataView} pre-shared key that must either be
+       a buffer (that will be interpreted as hexadecimally ecoded) or null to
+       stop the negotiation process.
+    When negotiating TLS-PSK (pre-shared keys), this function is called
+    with the identity provided by the client.
+    If the return value is `null` the negotiation process will stop and an
+    "unknown_psk_identity" alert message will be sent.
+    If the server wishes to hide the fact that the PSK identity was not known,
+    callback must provide some random data as psk, so the connection will fail
+    with "decrypt_error" before it will be finished completely.
+    Note that PSK ciphers are disabled by default, and using TLS-PSK thus
+    requires explicitly specifying a cipher suite with the `ciphers` option.
+    Additionally, it may be necessary to disable `rejectUnauthorized` if a
```

Comment:
This is a bug. If openssl allows a client to have both a PSK cipher suite configured, and a non-PSK certificate-based authentication configured so the server can choose which one to use, then the connection should be secure. disabling rejectUnauthorized means the cert-based auth won't happen, and the connection will be insecure. I don't see how we can release this feature with a security hole like this.

If this is fundamentally a problem (bug?) with openssl, we can raise with them, and then perhaps work harder in this PR to make it not possible for a server to both request a client certificate, and enable PSK ciphers.

It is also possible that this is something that could be fixed in our code, where we are validating a cert when we should not be when PSK was used.

You [said](https://github.com/nodejs/node/pull/23188#discussion_r228850947) to @addaleax that you would investigate this further. Have you? If you can't figure out what's going on, I could take a shot at looking at this.

Reply:
Sorry, I forgot to remove this from here. Refer to https://github.com/nodejs/node/pull/23188#issuecomment-447900198 and 
https://github.com/nodejs/node/pull/23188/files#diff-f6e3a86962eaf0897ab59e88b418e64fR986
```
    It will be necessary to provide a custom [`tls.checkServerIdentity()`][]
    for the connection as the default one will try to check hostname/IP
    of the server against the certificate but that's not applicable for PSK
    because there won't be a certificate present.
```